legend-pure-code-compiled-core-relational/src/main/resources/core_relational/relational/pureToSQLQuery/pureToSQLQuery.pure

```diff
@@ -4987,11 +4996,22 @@ function meta::relational::functions::pureToSqlQuery::processProject(ids:String[
          let nonOlap = $newColumnsWithPossiblePKs->filter(c|!$c->instanceOf(WindowColumn));
 
          let possiblePKs = $possiblePKsWithUpdatedData.first.values->cast(@Alias).relationalElement->cast(@TableAliasColumn).column->map(c| pair($c.name, ^PathInformation(type = $c.type->meta::relational::metamodel::datatype::dataTypeToCompatiblePureType(), relationalType = $c.type)));
+         
+         let importDataFlowCols = if($state.importDataFlow == true && $state.importDataFlowAddFks == true,
+                                     |let mainTable = $merge.data->toOne().alias.relation();
+                                      let joinedFkTables = if($state.importDataFlowJoinedTablesByTable->isEmpty(), |[], |$state.importDataFlowJoinedTablesByTable->toOne()->get($mainTable).values);
+                                      let joinedFkCols = $joinedFkTables->toIndexed()->map(p|
+                                         let prefix = 'fk_j' + $p.first->toString() + '_';
+                                         $state.importDataFlowFksByTable->toOne()->get($p.second).values->map(c|
+                                            let columnAliases = $c->extractColumnAliasesInSelectQuery($p.second, $merge, $context, $extensions);
```

Comment:
I am not sure if this would work as expected when dealing with advanced queries like 'exists' filter, multiple joins to same target table for different properties etc..

It should be fine if simple queries are being targeted at the moment.

Reply:
Discussed offline that the new impl will work for multiple joins to same target table. Included test for it as well.